### DISCUSSION TITLE
Performance improvement: consider inlining partials

### DISCUSSION THREAD

**Discussion by searls** [josefarias/hotwire_combobox#194] [Ideas]
Because performance optimizations to this library could add up to a really big impact for user's applications, I think it might be worth looking at trading off some file organization for simplicity by inlining the partials that hotwire_combobox renders.

I have a page with a dozen or more comboboxes, and it's hard not to notice that my development console is pretty full of this same stack of partials on repeat:

```
  Rendered /Users/justin/.rbenv/versions/3.3.3/lib/ruby/gems/3.3.0/gems/hotwire_combobox-0.3.1/app/views/hotwire_combobox/combobox/_hidden_field.html.erb (Duration: 0.0ms | GC: 0.0ms)
  Rendered /Users/justin/.rbenv/versions/3.3.3/lib/ruby/gems/3.3.0/gems/hotwire_combobox-0.3.1/app/views/hotwire_combobox/combobox/_input.html.erb (Duration: 0.0ms | GC: 0.0ms)
  Rendered /Users/justin/.rbenv/versions/3.3.3/lib/ruby/gems/3.3.0/gems/hotwire_combobox-0.3.1/app/views/hotwire_combobox/_pagination.html.erb (Duration: 0.0ms | GC: 0.0ms)
  Rendered /Users/justin/.rbenv/versions/3.3.3/lib/ruby/gems/3.3.0/gems/hotwire_combobox-0.3.1/app/views/hotwire_combobox/combobox/_paginated_listbox.html.erb (Duration: 0.1ms | GC: 0.0ms)
  Rendered /Users/justin/.rbenv/versions/3.3.3/lib/ruby/gems/3.3.0/gems/hotwire_combobox-0.3.1/app/views/hotwire_combobox/combobox/_dialog.html.erb (Duration: 0.1ms | GC: 0.0ms)
  Rendered /Users/justin/.rbenv/versions/3.3.3/lib/ruby/gems/3.3.0/gems/hotwire_combobox-0.3.1/app/views/hotwire_combobox/_component.html.erb (Duration: 2.4ms | GC: 0.0ms)
```

It's not likely to be a _major_ resource hog, but because form elements tend not to be as cacheable as other hierarchies of view partials, I wonder if @josefarias might be open to a PR inlining these into a single partial, assuming it wouldn't introduce significant code duplication

**Comment by josefarias** [comment]
Thanks @searls! Yeah I've no problem reducing the number of partials here. Some of these are called from the library helpers to render bits and pieces. But we could either keep those partials (which shouldn't be very many) or completely replace with helper calls.

There's a ton of moving parts — so it'd be interesting to see if we can get them all in a single template without making things too hard to change. Just so one could get a complete picture of the combobox sub-components at a glance. That'd be great but maybe more trouble than it's worth.

**Reply by josefarias** [reply]
done!

**Reply by searls** [reply]
Woot! Great job!
